TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -6,14 +6,22 @@ class BenchmarkParser extends Parser {
 
     static getBenchmarkInfo( testName ) {
         let result, benchmarkName, benchmarkVariant;
+
+        /* For PerfNext, benchmark name is allowed to have multiple underscores but variant shouldn't have underscores 
+         * in order for TRSS to parse the job. For TKG, we'll be good since we don't specify variant name anyway.
+         */
         if (testName) {
             if ((result = testName.split("_")) != null) {
-                if (result.length > 2) {// this works for benchmarks: ILOG_WODM, SOABBenh_10k10k, and maybe more in future.
-                    benchmarkName = result[0] + "_" + result[1];
-                    benchmarkVariant = result[2];
-                } else {
-                    benchmarkName = result[0];
-                    benchmarkVariant = result[1];
+                if (isArray(result) && result.length > 0) {
+                    benchmarkVariant = result[result.length - 1];
+                    let i;
+                    for (i = 0; i < result.length - 1; i++) {
+                        if (benchmarkName == null && i == 0) {
+                            benchmarkVariant = result[0];
```

Comment:
Slight confused regarding these lines. Why can't we start with `i=1` and get rid of `if...else`?